cmd/core/pdr_creation_context.go

```diff
@@ -137,6 +145,22 @@ func (pcc *PDRCreationContext) getFTEID(seID uint64, pdrID uint32) (uint32, erro
 	return allocatedTeid, nil
 }
 
+func (pcc PDRCreationContext) getIP(seID uint64) (net.IP, error) {
+	if pcc.ResourceManager == nil {
+		return nil, errors.New("resource manager is nil")
+	} else {
+		if pcc.ResourceManager.IPAM == nil {
+			return nil, errors.New("IPAM manager is nil")
+		}
+	}
+	allocatedIP, err := pcc.ResourceManager.IPAM.AllocateIP(seID)
+	if err != nil {
+		log.Error().Msgf("AllocateIP err: %v", err)
```

Comment:
I suppose it's enough to return formatted error. Log message will be written in calling code